src/mc_test_circuits/cert.rs

```diff
@@ -71,6 +86,23 @@ fn enforce_cert_inputs_gadget<CS: ConstraintSystemAbstract<FieldElement>>(
             cs.ns(|| format!("expected_cert_data_hash_must_be_eq_actual_{}", i)),
             &Boolean::Constant(true),
         )?;
+
+    }
+
+    if sc_prev_hash_g.is_some() {
+        let prev_hash = sc_prev_hash_g.unwrap();
+        let expected_prev_hash = expected_sc_prev_hash_g.unwrap();
+
+        for i in 0..remaining_constraints {
```

Comment:
We must update the calculations above, otherwise it does never enter in this for loop

Reply:
Makes sense, thanks, commit 16b63d9187a6cffded5290597e8e23ecee799ee5 should fix the size of the circuit.